frontend/src/_helpers/utils.js

```diff
@@ -227,3 +227,18 @@ export function validateWidget({ validationObject, widgetValue, currentState, cu
     validationError,
   };
 }
+
+export const getDynamicText = (text, currentState) => {
```

Comment:
The `resolveReferences` function will take care of resolving every code expression inside the string. There is no need to separate them out from scratch. I don't feel this new function is required.